### ISSUE TITLE
Underscores in parameters inconsistency

### ISSUE THREAD

**Issue by jeteve** [apiaryio/drafter#445]
Let's say I have:

```
## Bla [/bla{?some_param}]

+ Parameters
     + `some_param`:...

```
This creates a

```
'parameter '`some_param`' not found within the URL template'
```

semantic error.

Now if I do:

```
## Bla [/bla{?`some_param`}]
```
This creates a 'URI expression contains invalid character' error.

So either ways, I'm screwed. And it makes me very sad APIary. Everyone says you're good. Probably they don't use parameters with underscores in them.

I think the issue comes from the fact the validator considers '`some_param`' to be the parameter, not just 'some_param' without the backticks.

As far as I understand, backticks are just a way to define a string containing reserved characters in MSON, so surely they shouldn't be included in the parsed token.

**Comment by w-vi** [comment]
Thanks a lot for reporting this issue, MSON based Parameters are not yet adopted too much so bugs are very likely there as you have found.

**Comment by jeteve** [comment]
Thanks for the answer :) To be fair, it looks like a bug in whatever MSON parser you're using :)

**Comment by w-vi** [comment]
Indeed it is and I will push a fix soon :)

**Comment by kylef** [comment]
This was fixed in Drafter 3.2.3 and has been propagated to Apiary for some time now.
